game-core/src/main/java/games/strategy/engine/data/GameParser.java

```diff
@@ -1254,48 +1252,22 @@ private Attachable findAttachment(final Element element, final String type) thro
     return returnVal;
   }
 
-  private static String capitalizeFirstLetter(final String input) {
-    char first = input.charAt(0);
-    first = Character.toUpperCase(first);
-    return first + input.substring(1);
-  }
-
   private ArrayList<Tuple<String, String>> setValues(final IAttachment attachment, final List<Element> values)
       throws GameParseException {
+    final Map<String, Function<IAttachment, AttachmentProperty<?>>> attachmentMap = attachment.getAttachmentMap();
     final ArrayList<Tuple<String, String>> options = new ArrayList<>();
     for (final Element current : values) {
       // find the setter
       final String name = current.getAttribute("name");
       if (name.length() == 0) {
         throw newGameParseException("Option name with 0 length");
       }
-      final Method setter;
-      try {
-        setter = attachment.getClass().getMethod("set" + capitalizeFirstLetter(name), SETTER_ARGS);
-      } catch (final NoSuchMethodException nsme) {
-        throw newGameParseException(String.format(
-            "The following option name of %s of class %s are either misspelled"
-                + " or exist only in a future version of TripleA. Setter: %s",
-            attachment.getName(), attachment.getClass().getSimpleName(), name));
-      }
       // find the value
       final String value = current.getAttribute("value");
       final String count = current.getAttribute("count");
-      final String itemValues;
-      if (count.length() > 0) {
-        itemValues = count + ":" + value;
-      } else {
-        itemValues = value;
-      }
-      // invoke
-      try {
-        setter.invoke(attachment, itemValues);
-      } catch (final IllegalAccessException iae) {
-        throw newGameParseException(
-            "Setter not public. Setter:" + name + " Class:" + attachment.getClass().getName(), iae);
-      } catch (final InvocationTargetException ite) {
-        throw newGameParseException("Error setting property:" + name, ite);
-      }
+      final String itemValues = (count.length() > 0 ? count + ":" : "") + value;
+      attachmentMap.get(name).apply(attachment).setValue(itemValues);
```

Comment:
At least some of your tests are failing because the map XMLs under _src/test/resources_ have the same problem you fixed in all the map repos: capitalized option names (e.g. `CarrierCost` vs. `carrierCost`).

Note that you should probably guard against the map `get()` call returning `null` for an unknown property name.  Throwing a `GameParseException` would be appropriate in that case.

Reply:
@RoiEXLab Bumping the recommendation to explicitly check for the existence of an entry in `attachmentMap`:

```diff
diff --git a/game-core/src/main/java/games/strategy/engine/data/GameParser.java b/game-core/src/main/java/games/strategy/engine/data/GameParser.java
index 42a0ff5aa..6af0f70fd 100644
--- a/game-core/src/main/java/games/strategy/engine/data/GameParser.java
+++ b/game-core/src/main/java/games/strategy/engine/data/GameParser.java
@@ -1266,7 +1266,11 @@ public final class GameParser {
       final String value = current.getAttribute("value");
       final String count = current.getAttribute("count");
       final String itemValues = (count.length() > 0 ? count + ":" : "") + value;
-      attachmentMap.get(name).apply(attachment).setValue(itemValues);
+      final Function<IAttachment, AttachmentProperty<?>> propertyFunction = attachmentMap.get(name);
+      if (propertyFunction == null) {
+        throw newGameParseException("Missing property definition for option: " + name);
+      }
+      propertyFunction.apply(attachment).setValue(itemValues);
 
       options.add(Tuple.of(name, itemValues));
     }
```